hnn_core/tests/test_network.py

```diff
@@ -79,6 +80,22 @@ def test_spikes():
         spikes = Spikes(times=[[2.3456, 7.89]], gids=spikegids,
                         types=spiketypes)
 
+    with pytest.raises(TypeError, match="spike_types should be str, "
+                                        "list, dict, or None"):
+        spikes.plot_hist(spike_types=1, show=False)
+
+    with pytest.raises(TypeError, match=r"spike_types\[ev\] must be a list\. "
+                                        r"Got int\."):
+        spikes.plot_hist(spike_types={'ev': 1}, show=False)
+
+    with pytest.raises(ValueError, match=r"Elements of spike_types must map to"
+                       r" mutually exclusive input types\. L2_basket is found"
```

Comment:
do you actually need to escape the stop?

Reply:
Unfortunately yes... since the error message has conditional formatting there are some rather cryptic regex matching errors. Easiest way to handle them in general is to convert to raw strings, hence the escapes.

Reply:
I see, I typically just put in the first few words to avoid such hassles. You pretty much guarantee a unique match given that we have only a few error messages. But this is nice too as it's more explicit

Reply:
Didn't realize it didn't have to be an exact match! Good to know for the future.